lib/fizzy/parser.cpp

```diff
@@ -531,11 +547,26 @@ Module parse(bytes_view input)
     if (!module.elementsec.empty() && !module.has_table())
         throw validation_error("element section encountered without a table section");
 
+    const auto total_global_count = module.get_global_count();
+    for (const auto& global : module.globalsec)
+    {
+        validate_constant_expression(global.expression, module);
+
+        // Wasm spec section 3.3.7 constrains initialization by another global to const imports only
+        // https://webassembly.github.io/spec/core/valid/instructions.html#expressions
+        if (global.expression.kind == ConstantExpression::Kind::GlobalGet &&
+            global.expression.value.global_index >= module.imported_globals_mutability.size())
+        {
+            throw validation_error(
```

Comment:
Also uses parenthesis. It seems we are inconsistently using parentheses and curly brackets.